### PR TITLE
Added return and server configurations

### PR DESCRIPTION
**Pull request by anirudh-hegde**
**Notes for Reviewers**

As return statement, it allows deferred functions to run before exiting.
And also added configure file type and its path.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — COMMENTED** [review]
Thanks, @anirudh-hegde. 

Note: It's generally helpful, if you create an issue, so that 1) you avoid potentially putting in work that is unwanted/invalid and 2) acclimate maintainers to the problem at-hand quickly, so the PR reviews are expedient.

**Comment by anirudh-hegde** [comment]
> Thanks, @anirudh-hegde.
> 
> Note: It's generally helpful, if you create an issue, so that 1) you avoid potentially putting in work that is unwanted/invalid and 2) acclimate maintainers to the problem at-hand quickly, so the PR reviews are expedient.

Yeah, I will remember from next time.

**Comment by vishalvivekm** [comment]
@anirudh-hegde 
 Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

File: server/cmd/main.go
Comment (MUzairS15) [inline server/cmd/main.go:61]: Why is it required? viper reads from the file specified in line 63.

**Review by MUzairS15 — CHANGES_REQUESTED** [review]
@anirudh-hegde The defer functions run after the main function exits.
Why it is necessary?

File: server/cmd/main.go
Comment (anirudh-hegde) [reply server/cmd/main.go:61]: If configuration data can come from multiple sources also including environment variables, then these functions are required

**Comment by anirudh-hegde** [comment]
> @anirudh-hegde Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

Can you please tell what is the agenda for the meeting?

File: server/cmd/main.go
Comment (MUzairS15) [reply server/cmd/main.go:61]: AutomaticEnv?

Comment (anirudh-hegde) [reply server/cmd/main.go:61]: > AutomaticEnv?

This is to automatically override and read env variables from the configuration file

**Comment by leecalcote** [comment]
Merge conflict.

@anirudh-hegde the agenda would be you presenting your work and receiving feedback on it. It's optional. It's encouraged. Faster feedback loop. Higher fidelity communication and relationships.

**Comment by anirudh-hegde** [comment]
> Merge conflict.
> 
> @anirudh-hegde the agenda would be you presenting your work and receiving feedback on it. It's optional. It's encouraged. Faster feedback loop. Higher fidelity communication and relationships.

Yeahh

**Review by leecalcote — CHANGES_REQUESTED** [review]
@anirudh-hegde, after fixing the merge conflict, please rebase/resync your fork. Only your commits should be included in this PR. Ping when ready for review.

**Comment by anirudh-hegde** [comment]
Hey @leecalcote can you please review

File: server/cmd/main.go
Comment (leecalcote) [reply server/cmd/main.go:61]: @MUzairS15  is this a desired change? Overkill, inconsequential? Should @anirudh-hegde be redirected to work on an existing issue or is this unacceptable change?

**Comment by leecalcote** [comment]
@anirudh-hegde you still need to re-sync your fork. Only your commit should be included in this PR.

**Comment by leecalcote** [comment]
@anirudh-hegde you still need to re-sync your fork. Only your commit should be included in this PR.

**Comment by anirudh-hegde** [comment]
I synced my fork

**Comment by leecalcote** [comment]
I’m still seeing other peoples’ commits…

**Comment by leecalcote** [comment]
Let’s give you one more try at this and close it out. If you don’t get it on your next attempt.

**Pull request closed without merging**
